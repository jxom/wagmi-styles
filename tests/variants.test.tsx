import React from 'react';
import { View } from 'react-native';
import { render } from '@testing-library/react-native';

import {
  themeA,
  themeB,
  ThemeProvider,
  style,
  styleVariants,
  useStyleVariants,
} from './theme';

describe('variants', () => {
  const variantsStyleRef = styleVariants({
    small: {
      width: 100,
      height: 100,
    },
    large: {
      width: 300,
      height: 300,
    },
  });

  function Variants() {
    const variantsStyle = useStyleVariants(variantsStyleRef);
    return (
      <>
        <View style={variantsStyle.small} />
        <View style={variantsStyle.large} />
      </>
    );
  }

  test('variants style', () => {
    const { toJSON } = render(
      <ThemeProvider theme={themeA}>
        <Variants />
      </ThemeProvider>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  ////////////////////////////////////////////////////////////////////////////////

  const size = style((vars) => ({
    height: vars.sizes.small,
    width: vars.sizes.small,
  }));
  const spacing = style({ padding: 16, margin: 8 });
  const variantsWithCompositionStyleRef = styleVariants((vars) => ({
    primary: [size, spacing, { backgroundColor: vars.colors.primary }],
    secondary: [
      size,
      spacing,
      {
        backgroundColor: vars.colors.secondary,
      },
    ],
  }));

  function VariantsWithComposition() {
    const variantsStyle = useStyleVariants(variantsWithCompositionStyleRef);
    return <View style={variantsStyle.primary} />;
  }

  [themeA, themeB].forEach((theme) => {
    test('variants w/ composition style', () => {
      const { toJSON } = render(
        <ThemeProvider theme={theme}>
          <VariantsWithComposition />
        </ThemeProvider>
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
