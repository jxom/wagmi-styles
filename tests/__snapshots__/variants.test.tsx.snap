// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`variants variants style 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "height": 100,
          "width": 100,
        },
      ]
    }
  />,
  <View
    style={
      Array [
        Object {
          "height": 300,
          "width": 300,
        },
      ]
    }
  />,
]
`;

exports[`variants variants w/ composition style 1`] = `
<View
  style={
    Array [
      Object {
        "height": 100,
        "width": 100,
      },
      Object {
        "margin": 8,
        "padding": 16,
      },
      Object {
        "backgroundColor": "blue",
      },
    ]
  }
/>
`;

exports[`variants variants w/ composition style 2`] = `
<View
  style={
    Array [
      Object {
        "height": 200,
        "width": 200,
      },
      Object {
        "margin": 8,
        "padding": 16,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
/>
`;
